### DISCUSSION TITLE
How to render only inlines?

### DISCUSSION THREAD

**Discussion by ducaale** [thephpleague/commonmark#656] [Q&A] [answered]
is there a method for inline rendering ?
example:

$converter = new CommonMarkConverter();
$converter->convertToHtml('hello'); 

to produce `hello` instead of `<p>hello</p>`

**Comment by colinodell** [comment]
You could create a custom renderer for `Paragraph` blocks which only renders its inlines and not the `<p>` tags:

```php
class InlineOnlyRenderer implements BlockRendererInterface
{
    public function render(AbstractBlock $block, ElementRendererInterface $htmlRenderer, $inTightList = false)
    {
        return $htmlRenderer->renderInlines($block->children());
    }
}
```

To use it:

```php
// Create the standard environment with all the default parsers, renderers, etc.
$environment = Environment::createCommonMarkEnvironment();
// Override the default Paragraph renderer with the one we just created
$environment->addBlockRenderer(Paragraph::class, new InlineOnlyRenderer());

// Instantiate the converter with the custom Environment we just configured
$converter = new CommonMarkConverter([], $environment);

// Convert to HTML
echo $converter->convertToHtml('**Hello World!**');
```

I hope that helps!

**Comment by colinodell** [comment, marked as answer]
We now have an [Inlines-Only Extension](https://commonmark.thephpleague.com/extensions/inlines-only/) which solves this use case.

**Reply by ghost** [reply]
@colinodell Thanks for the extension.

Is there any way to remove or make `CommonMark\Node\Inline\Image::class` optional.

**Reply by colinodell** [reply]
You have three options here:

1. Add an event listener that iterates the AST and removes `Image::class` instances from the `Document` before rendering happens
2. Add an additional custom renderer for `Image::class` with a high priority and have it always return empty strings
3. Create your own copy of  [InlinesOnlyException](https://github.com/thephpleague/commonmark/blob/2.0/src/Extension/InlinesOnly/InlinesOnlyExtension.php) without the `ImageRenderer`

All of them will have the same result where the parser will still locate images in the Markdown but prevent them from showing up in the output.

**Reply by ghost** [reply]
Another option to me is to just use `preg_replace` to the converted html. Thanks!
